core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -353,13 +355,26 @@ public void close(Throwable cause) {
         }
 
         private void close(@Nullable Throwable cause, boolean cancel) {
+            if (closed) {
+                return;
+            }
+            closed = true;
             state = State.DONE;
             cancelTimeoutOrLog(cause, cancel);
             if (ctx != null) {
-                if (cause == null) {
-                    ctx.request().abort(ResponseCompleteException.get());
-                } else {
-                    ctx.request().abort(cause);
+                final HttpRequest request = ctx.request();
+                assert request != null;
+                if (cause != null) {
+                    request.abort(cause);
+                    return;
+                }
+                if (ctx.requestAutoAbortDelayMillis() == 0 || !request.isOpen()) {
```

Comment:
Q: Do we need to abort the request even if the request was closed?

Reply:
No, we shouldn't. 😓 Let me remove the condition. Thanks!